core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -24,20 +24,14 @@
 import java.util.Set;
 import org.apache.iceberg.encryption.EncryptedOutputFile;
 import org.apache.iceberg.events.CreateSnapshotEvent;
-import org.apache.iceberg.exceptions.CommitFailedException;
 import org.apache.iceberg.exceptions.RuntimeIOException;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.util.CharSequenceSet;
 
-/**
- * {@link AppendFiles Append} implementation that adds a new manifest file for the write.
- *
- * <p>This implementation will attempt to commit 5 times before throwing {@link
```

Comment:
why not rather update the comment and refer to the table property that is being used for retries: `TableProperties#COMMIT_NUM_RETRIES_DEFAULT`? 
In your PR description you mention that retries aren't done for `FastAppend`, but retries are done for all subclasses of `SnapshotProducer`.
Also if we update this javadoc, then we should align this for other subclasses of `SnapshotProducer` as well

Reply:
Thanks for taking a look, @nastra!
Frankly, I don' really prefer redundant comments. I don't see why we'd want to add the same comment for all the subclasses of SnapshotProducer. First, as said in my description retries don't happen on that level of abstraction. Let's say for some reason the retry logic or the name of the config changes in the future. Then we have to remember that there is a comment about this in all the subclasses that we have to change now too.

What if I add a comment about the retries to SnapshotProducer itself? That's the level of abstraction where that logic happens, would be more suitable and we wouldn't have redundant comments in the subclasses.